tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,179 @@
+	g "maistra.io/istio-operator/pkg/util/tests/ginkgo"
+	"maistra.io/istio-operator/pkg/util/tests/kubectl"
+	resourcecondition "maistra.io/istio-operator/pkg/util/tests/types"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(120 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+	var (
+		resourceAvailable = resourcecondition.Condition{
+			Type:   "Available",
+			Status: "True",
+		}
+		resourceReconcilied = resourcecondition.Condition{
+			Type:   "Reconciled",
+			Status: "True",
+		}
+
+		resourceReady = resourcecondition.Condition{
+			Type:   "Ready",
+			Status: "True",
+		}
+	)
+
+	Describe("installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("default helm manifest are applied", func() {
+			BeforeAll(func() {
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+				g.Success("Deploying Operator using default helm charts located in /chart folder")
+				Eventually(deployOperator).Should(Succeed(), "Operator failed to be deployed; unexpected error")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetResourceCondition).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable))
+				g.Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=istio-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				g.Success("Istio-operator pod is Running")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		AfterAll(func() {
+			// Cleanup the operator at the end of the test
+			Eventually(undeployOperator).Should(Succeed(), "Operator failed to be deleted; unexpected error")
+			g.Success("Operator deployment is deleted")
+		})
+
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created; unexpected error")
+				})
+
+				AfterAll(func() {
+					g.Success("Cleaning up")
+					Eventually(kubectl.DeleteNamespace).
+						WithArguments(controlPlaneNamespace).
+						Should(Succeed(), "Namespace failed to be deleted; unexpected error")
+
+					Eventually(kubectl.CheckNamespaceExist).
+						WithArguments(controlPlaneNamespace).
+						Should(MatchError(kubectl.ErrNotFound), "Namespace should not exist; unexpected error")
+					g.Success("Cleanup done")
+				})
+
+				When("the resource is created", func() {
+					It("deploys the Istio CR", func() {
```

Comment:
Is `It` appropriate here? Deploying the Istio CR isn't an assertion, but a setup setp.

Reply:
If is a setup should be in the before step